src/plugin/ipc/file/fileconnection.cpp

```diff
@@ -342,7 +342,9 @@ FileConnection::refill(bool isRestart)
 
       if (stat(_path.c_str(), &statbuf) == 0 && S_ISREG(statbuf.st_mode)) {
         if (statbuf.st_size > _st_size &&
-            ((_fcntlFlags & O_WRONLY) || (_fcntlFlags & O_RDWR))) {
+            ((_fcntlFlags & O_APPEND) ||
```

Comment:
Please change the order of flags. It should be O_WRONLY, O_RDWR, and then O_APPEND. In a weird setting (O_APPEND + O_RDONLY), truncate would fail.

Reply:
I meant if a file was open in an append mode and with read-write permission. However, the application chooses to change the access mode.

Reply:
We are reading flags from fcntl call so should have the right values. Even if there were some weird combination, it would be okay to truncate the file.